prow/io/opener.go

```diff
@@ -137,15 +137,18 @@ var ErrNotFoundTest = fmt.Errorf("not found error which should only be used in t
 
 // IsNotExist will return true if the error shows that the object does not exist.
 func IsNotExist(err error) bool {
-	if os.IsNotExist(err) || err == storage.ErrObjectNotExist {
+	if os.IsNotExist(err) {
 		return true
 	}
-	if err == ErrNotFoundTest {
+	if errors.Is(err, ErrNotFoundTest) {
 		return true
 	}
 	if errors.Is(err, os.ErrNotExist) {
 		return true
 	}
+	if errors.Is(err, storage.ErrObjectNotExist) {
+		return true
+	}
 	return gcerrors.Code(err) == gcerrors.NotFound
```

Comment:
Its not possible to test this because the error type that is used in the check is in an `internal` package and there is no constructor for it